pkg/virt-controller/watch/vm.go

```diff
@@ -2375,3 +2386,68 @@ func (c *VMController) applyDevicePreferences(vm *virtv1.VirtualMachine, vmi *vi
 	}
 	return nil, nil
 }
+
+func (c *VMController) handleInterfaceRequests(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if len(vm.Status.InterfaceRequests) == 0 {
+		return nil
+	}
+
+	interfaceMap := map[string]virtv1.Interface{}
+	if vmi != nil {
+		interfaceMap = netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
+	}
+	for i, ifaceRequest := range vm.Status.InterfaceRequests {
+		vm.Spec.Template.Spec = *controller.ApplyNetworkInterfaceRequestOnVMISpec(
+			&vm.Spec.Template.Spec, &vm.Status.InterfaceRequests[i])
+
+		if vmi == nil || vmi.DeletionTimestamp != nil {
+			continue
+		}
+
+		if ifaceRequest.AddInterfaceOptions != nil {
+			if _, exists := interfaceMap[ifaceRequest.AddInterfaceOptions.InterfaceName]; exists {
+				continue
+			}
+
+			if err := c.clientset.VirtualMachineInstance(vmi.Namespace).AddInterface(context.Background(), vmi.Name, ifaceRequest.AddInterfaceOptions); err != nil {
```

Comment:
virt-api is an API wrapper/proxy/add-on of the system. It does not seem to me to be a core component. E.g. everything should continue working and not get stuch if it is down.

If a core component depends on it, especially in a blocking operation, it may block the core system flow and not only cause a local failure (which reconciles).
Also, the notion of reconcile is not a reason to have something like this IMO. Yes, in some cases the reconcile will recover, but if we can remove the dependency completely, we will just not be in that scenario in the first place.